maintenance/health/servicehealthcheck/background_healthchecker.go

```diff
@@ -0,0 +1,89 @@
+package servicehealthcheck
+
+import (
+	"context"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/pace/bricks/maintenance/log"
+)
+
+// BackgroundHealthCheck is an interface that, when implemented, will run the
+// HealthCheck in the background, and cache the result, that will be fetched by the HealthHandler
+type BackgroundHealthCheck interface {
+	HealthCheck
+	Interval() time.Duration
+}
+
+// BackgroundHealthCheckMaxWait allows to use a custom amount of time to wait before failing the health check
+type BackgroundHealthCheckMaxWait interface {
+	MaxWait() time.Duration
+}
+
+// registerBackgroundHealthCheck will run the HealthCheck in the background at every given interval.
+// The returned backgroundStateHealthChecker is then used to query this cached state when needed.
+func registerBackgroundHealthCheck(name string, bhc BackgroundHealthCheck) *backgroundStateHealthChecker {
+	var bgState ConnectionState
+
+	go func() {
+		s := log.Sink{Silent: true}
+		ctx := log.ContextWithSink(context.Background(), &s)
```

Comment:
Have you tried running this code without any initialization of any logger. Because I think if this runs isolated from any other contexts if no logger is attached to the context then no logging actually happens. That is at least my observation when I forget to attach anything to the context regarding logging.

But I am blissfully unaware of the internals of our zerolog wrapper.

Reply:
Yep, no need for it here as it uses the `context.Background()`.  

I copied this from the other function that had this, but that was called from an `http.Handler` with an active logger in the context so it made sense to mute it.